docroot/modules/custom/va_gov_backend/va_gov_backend.module

```
@@ -328,3 +340,104 @@ function va_gov_backend_pathauto_alias_alter(&$alias, array &$context) {
+        // Store reference in field_listing.
+        $entity->set('field_listing', NULL);
+        $entity->field_listing[] = ['target_id' => $office];
+      }
+      else {
+        // Find event_listing by office or health_care_region_page id.
+        $result = Drupal::entityQuery('node')
+          ->condition('type', 'event_listing')
+          ->condition('field_office', $office)
+          ->execute();
+
+        _va_gov_backend_set_field_listing_value($entity, $entity_listing_mapping, reset($result));
+      }
+      break;
+
+    // Story; News release; Publication.
+    case 'news_story':
+    case 'press_release':
+    case 'outreach_asset':
+      // Check if the entity reference is an office or health_care_region_page.
+      if ($entity_ref && in_array($entity_ref->bundle(), ['office', 'health_care_region_page'])) {
+        // Find *_listing node by office or health_care_region_page id.
+        $result = Drupal::entityQuery('node')
+          ->condition('type', $entity_listing_mapping[$entity->bundle()])
+          ->condition('field_office', $office)
+          ->execute();
+
+        _va_gov_backend_set_field_listing_value($entity, $entity_listing_mapping, reset($result));
+      }
+      break;
+  }
+}
+
+/**
+ * Sets field_listing value or throws an exception for an empty value.
+ *
+ * @param \Drupal\Core\Entity\EntityInterface $entity
+ *   Entity.
+ * @param array $entity_listing_mapping
+ *   Mapping that specifies how content type maps to corresponding
+ *   "listing" content type.
+ * @param int|null $id
+ *   NID of the node of type "listing".
+ *
+ * @throws \Exception
+ */
+function _va_gov_backend_set_field_listing_value(EntityInterface &$entity, array $entity_listing_mapping, int $id = NULL) {
+  if (!is_int($id) || $id == 0) {
+    // Listing corresponding to field_office, was not found. Fail silently.
+    // Give a warning messages to the user and log in dblog.
```

Giggle:  Fail silently. ... Give a warning